src/pyaro/timeseries/Filter.py

```diff
@@ -614,7 +614,7 @@ def envelope(self) -> tuple[datetime, datetime]:
         end = datetime.min
         for s, e in self._start_include + self._startend_include + self._end_include:
             start = min(start, s)
-            end = max(end, s)
+            end = max(end, e)
```

Comment:
Is this the cleanest way to set the `start`/`end`
Also, why is does `start` depends on `self._end_include` and `end` depends on `self._start_include`?

maybe [itertools.chain] can improve the readability

``` python
from itertools import chain
[...]
        start = min(s for s, e in chain(self._start_include, self._startend_include))
        end = max(e for s, e in chain(self._end_include, self._startend_include))
```

[itertools.chain]: https://docs.python.org/3/library/itertools.html#itertools.chain

Reply:
I don't think the chain-version is very readable here.

Reply:
> I don't think the chain-version is very readable here.

Yes, at the cost of creating new lists this could be written on a simpler way
``` python
        if self._start_include or self._startend_include:
            start = min(s for s, e in self._start_include + self._startend_include)
        else:
            start = datetime.max

        if self._end_include or self._startend_include:
            end = max(e for s, e in self._end_include + self._startend_include)
        else:
            end = datetime.min
```

In any case, my question still stands 

>> Also, why is does start depends on self._end_include and end depends on self._start_include?

Reply:
The tuple values of (start, end) are the time-ranges of the filter, while start_include/end_include refer to measurements starttime/endtime.